sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,198 @@
+from abc import ABCMeta, abstractmethod
+from copy import deepcopy
+
+import numpy as np
+
+from ..base import clone
+from ..base import ClassifierMixin, RegressorMixin, TransformerMixin
+from ..base import is_classifier, is_regressor
+from ..base import MetaEstimatorMixin
+
+from .base import _parallel_fit_estimator
+
+from sklearn.externals.joblib import Parallel, delayed
+from sklearn.externals.six import with_metaclass
+from sklearn.externals.six import string_types
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model import LinearRegression
+
+from ..model_selection import cross_val_predict
+from ..model_selection import check_cv
+
+from ..utils import check_random_state
+from ..utils.metaestimators import _BaseComposition
+from ..utils.metaestimators import if_delegate_has_method
+from ..utils.validation import has_fit_parameter
+from ..utils.validation import check_is_fitted
+
+
+class BaseStacking(with_metaclass(ABCMeta, _BaseComposition,
+                                  MetaEstimatorMixin, TransformerMixin)):
+
+    def __init__(self, estimators=None, meta_estimator=None, cv=3,
+                 method_estimators='auto', n_jobs=1, random_state=None,
+                 verbose=0):
+        self.estimators = estimators
+        self.meta_estimator = meta_estimator
+        self.cv = cv
+        self.method_estimators = method_estimators
```

Comment:
I believe that `method_estimators` is not obvious name. `output_confidence`, `metafeatures` or something like that could be more targeted.  In case of classification, if `output_confidence` is `True` the the level 1 estimator should be trained on the (cross-validated or not) outputs of `predict_proba` or `decision_function` of level 0 estimators else the level 1 estimator should be trained on prediction outputs. Another option could be a parameter like `metafeatures` with several options like:
* predictions
* confidences
* cv_predictions
* cv_confidences

Of course we could use both metafeatures and the original space to train the level 1 estimator. 

As for the confidence outputs an option could be to aggregate the outputs using a descriptive statistic like `mean` or `median`. So the dimensionality of the metafeatures will be the same as the number of classes.

Reply:
What the use case in which we want to avoid the CV when fitting the `final estimator`.
`method_estimators` is the closest keywords that I found to be in accordance with the `method` keyword in the other estimator (in grid-search I think)

Reply:
How about `predict_method`?